packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -32,27 +35,48 @@ class DependencyExtractionWebpackPlugin {
 			options
 		);
 
-		/*
+		/**
 		 * Track requests that are externalized.
 		 *
 		 * Because we don't have a closed set of dependencies, we need to track what has
 		 * been externalized so we can recognize them in a later phase when the dependency
 		 * lists are generated.
+		 *
+		 * @type {Set<string>}
 		 */
 		this.externalizedDeps = new Set();
 
-		// Offload externalization work to the ExternalsPlugin.
+		/**
+		 * Should we use modules. This will be set later to match webpack's
+		 * output.module option.
+		 *
+		 * @type {boolean}
+		 */
+		this.useModules = false;
+
+		/**
+		 * Offload externalization work to the ExternalsPlugin.
+		 */
 		this.externalsPlugin = new webpack.ExternalsPlugin(
 			'window',
```

Comment:
This `'window'` parameter says that the externals are on the `window` global object. Makes sense only for the `useModules = false` mode. Does the parameter do something in `useModules = true` mode or is it completely ignored?

Reply:
This plugin reads the compiler options to know whether to use modules. We don't have access to the compiler when we create an instance of the plugin, only when webpack calls `apply`. At that time we update `useModules` and also set the externals plugin `type` appropriately before calling `apply` on it:

https://github.com/WordPress/gutenberg/blob/0b094345fcc7875c8504dcd4d0829e75ef6981c7/packages/dependency-extraction-webpack-plugin/lib/index.js#L142-L149

Reply:
Oh, that's insidious! Is there any way to postpone the creation of `this.externalsPlugin` until we know all the parameters?

Reply:
Sure, we can instantiate the externals plugin inside the `apply` method.

Reply:
I pushed this change.